src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/get_vehicle_data_request.cc

```diff
@@ -119,13 +119,30 @@ void GetVehicleDataRequest::on_event(const event_engine::Event& event) {
       hmi_apis::Common_Result::eType result_code =
           static_cast<hmi_apis::Common_Result::eType>(
               message[strings::params][hmi_response::code].asInt());
+      mobile_apis::Result::eType mobile_result_code =
+          GetMobileResultCode(result_code);
       bool result = PrepareResultForMobileResponse(
           result_code, HmiInterfaces::HMI_INTERFACE_VehicleInfo);
       std::string response_info;
       GetInfo(message, response_info);
-      result = result ||
-               ((hmi_apis::Common_Result::DATA_NOT_AVAILABLE == result_code) &&
-                (message[strings::msg_params].length() > 1));
+
+      auto data_not_available_with_params = [&result_code, &message]() {
+        if (hmi_apis::Common_Result::DATA_NOT_AVAILABLE != result_code) {
+          return false;
+        }
+
+        const auto& msg_params = message[strings::msg_params];
+        switch (msg_params.length()) {
+          case 0:
```

Comment:
i think it's better to avoid signed/unsigned comparisons

Reply:
@mked-luxoft fixed in 5db179eb26efd1a99c0178db3afb9d1e077cb383